src/terminal/parser/ut_parser/OutputEngineTest.cpp

```diff
@@ -2011,4 +2018,63 @@ class StateMachineExternalTest final
 
         pDispatch->ClearState();
     }
+
+    TEST_METHOD(TestSetClipboard)
+    {
+        auto dispatch = std::make_unique<StatefulDispatch>();
+        auto pDispatch = dispatch.get();
+        auto engine = std::make_unique<OutputStateMachineEngine>(std::move(dispatch));
+        StateMachine mach(std::move(engine));
+
+        mach.ProcessString(L"\x1b]52;;Zm9v\x07");
+        VERIFY_ARE_EQUAL(L"foo", pDispatch->_copyContent);
+
+        pDispatch->ClearState();
+
+        mach.ProcessString(L"\x1b]52;;Zm9vDQpiYXI=\x07");
+        VERIFY_ARE_EQUAL(L"foo\r\nbar", pDispatch->_copyContent);
+
+        pDispatch->ClearState();
+
+        mach.ProcessString(L"\x1b]52;s0;Zm9v\x07");
```

Comment:
nit: I might add comments explaining what each of these cases is - it's hard to see at a glance why `\x1b]52;s0;Zm9v\x07` should work and `\x1b]52;Zm9v\x07` doesn't, without having to refer to the spec.

Reply:
Explanations have been added to the tests.